src/Services/QueuedJobService.php

```diff
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
+            } else {
+                // Initial delay is missing - retry right away
+                $delay = 0;
+            }
+
+            $dateMessageSegment = 'immediately';
+
+            // Add a delay if configured, so we wouldn't retry the job immediately, but with some spread
+            // this lowers the chance of multiple jobs accessing the same data, thus preventing conflicts
+            if ($delay > 0) {
+                $scheduledTime = $now->getTimestamp() + $delay;
+                $jobDescriptor->StartAfter = $scheduledTime;
+
+                $formattedDate = DBDatetime::create_field('Datetime', $scheduledTime);
+                $dateMessageSegment = sprintf('after %s', $formattedDate->Rfc2822());
+            }
+
+            // Capture exception in the messages of the stuck job for better debug options
+            $message = sprintf('Automatic job retry - job will retry %s', $dateMessageSegment);
+
+            $this->copyDescriptorToJob($jobDescriptor, $job);
+
+            // Mark our retry attempt
+            $jobDescriptor->RetryCount += 1;
+
+            // Change the job status according to our map configuration
+            $jobDescriptor->JobStatus = $jobStatusMap[$jobDescriptor->JobStatus];
+
+            // Release the job lock, so it could be picked up again as well as expiry
+            $this->releaseJobLock($jobDescriptor);
+            $jobDescriptor->Expiry = null;
+
+            // This extension hook is useful not only to customise the message but also to integrate
+            // with another logging solution such as Graylog which provides the capability
+            // to see number of job retries over time as a visual chart
+            $this->extend('updateAutoRetryJobMessage', $jobDescriptor, $job, $message);
+
+            $job->addMessage($message);
+            $this->copyJobToDescriptor($job, $jobDescriptor);
+            $jobDescriptor->write();
+        }
+    }
+
+    /**
+     * Generate a randomised multiplier variance
+     */
+    protected function getJobRetryDelayMultiplier(float $multiplier, float $variance): float
+    {
+        // Variance is not in use or invalid
+        if (!$variance || $variance >= $multiplier) {
+            return $multiplier;
+        }
+
+        // Convert to integers for simpler calculation
+        $baseValue = (int) floor($multiplier * 100);
+        $offsetValue = (int) floor($variance * 100);
+
+        // This needs to be wrapped inside a separate non-private method so we can override it in
+        // situations such as tests as randomisation is not desirable in those cases
+        $multiplierKey = $this->getRandomValueFromRange($baseValue - $offsetValue, $baseValue + $offsetValue);
+
+        // Convert the final value back to float
+        return $multiplierKey / 100;
+    }
+
+    /**
+     * This functionality needs to be in a separate method so it can be overridden in unit tests
+     * where randomisation is not desirable
```

Comment:
```suggestion
     * where randomisation is not desirable
     * @internal
```
Since it's protected purely for the purpose of unit testing (and `private` methods can't be mocked), we should mark it `@internal` so it doesn't impact our commitments regarding [public API](https://docs.silverstripe.org/en/project_governance/public_api/).

Reply:
Will mark as internal but also noting that this can be used to override the randomisation in a project as well.